File: keiner_lindarte_aro/instructor_henry/mis_programas/practicas/python/clases_orientacion_objetos/7_clase_de_agosto_23/clase_abstracta.py
from abc import ABC, abstractmethod

class Animal(ABC):
    patas = 0
    promedio_de_vida = ""
    tipo_de_animal = ""
    comida = ""
    
    @abstractmethod
    def emitirsonido():
        raise "porfavor implementa este metodo"
    
    @abstractmethod
    def info():
        raise "porfavor implementa este metodo"
    

class Perro(Animal):
    patas = 4
    promedio_de_vida = "10-13 años"
    tipo_de_animal = "Mamifero"
    comida = "carne"
    
    def emitirsonido():
        print("Whuaauuuuuuu!!!")
    
    def info(self):
        print("El perro es un animal "+self.tipo_de_animal+" de "+str(self.patas)+" patas, con un promedio de vida de "+self.promedio_de_vida+" y su alimentacion es de "+self.comida+", su sonido caracteristico es:")
        self.emitirsonido
        
class Gato(Animal):
    patas = 4
    promedio_de_vida = "12-18 años"
    tipo_de_animal = "Mamifero"
    comida = "carne"
    
    def emitirsonido():
        print("Miaauuuuuu!!!")
    
    def info(self):
        print("El gato es un animal "+self.tipo_de_animal+" de "+str(self.patas)+" patas, con un promedio de vida de "+self.promedio_de_vida+" y su alimentacion es de "+self.comida+", su sonido caracteristico es:")
        self.emitirsonido
        
nuevo_perro = Perro()
nuevo_perro.info()

nuevo_gato = Gato()
nuevo_gato.info()

File: keiner_lindarte_aro/instructor_henry/mis_programas/java/clases_orientacion_objetos/clase_7_de_agosto_23/clase_abstracta.java
package ADSO2928707.portafolio_aprendices.keiner_lindarte_aro.instructor_henry.mis_programas.java.clases_orientacion_objetos.clase_7_de_agosto_23;

public class clase_abstracta {
    
    public static void main(String[] args) {
        
        Perro nuevo_perro = new Perro();
        nuevo_perro.info();
      
        Gato nuevo_gato = new Gato();
        nuevo_gato.info();

    }
}

abstract class Animal {

    int patas = 0;
    String promedio_de_vida = "";
    String tipo_de_animal = "";
    String comida = "";
    abstract void emitirsonido();
    abstract void info();
    
}

class Perro extends Animal{

    int patas = 4;
    String promedio_de_vida = "10-13 años";
    String tipo_de_animal = "Mamifero";
    String comida = "carne";
  
    void emitirsonido(){
      System.out.println("Whuaauuuuuuu!!!");
    }
  
    void info(){
        System.out.println("El perro es un animal "+tipo_de_animal+" de "+patas+" patas, con un promedio de vida de "+promedio_de_vida+" y su alimentacion es de "+comida+", su sonido caracteristico es:");
        emitirsonido();
    }
  
}

class Gato extends Animal{

    int patas = 4;
    String promedio_de_vida = "12-18 años";
    String tipo_de_animal = "Mamifero";
    String comida = "carne";
  
    void emitirsonido(){
      System.out.println("Miaauuuuuu!!!");
    }
  
    void info(){
        System.out.println("El gato es un animal "+tipo_de_animal+" de "+patas+" patas, con un promedio de vida de "+promedio_de_vida+" y su alimentacion es de "+comida+", su sonido caracteristico es:");
        emitirsonido();
    }
  
}

